xrpl/models/transactions/transaction.py

```diff
@@ -78,6 +72,13 @@ def _value_to_tx_json(value: XRPL_VALUE_TYPE) -> XRPL_VALUE_TYPE:
     return value
 
 
+def _auth_account_value_to_tx_json(value: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
Does this still need a special case?

Reply:
Looks like it does when I remove this special case:

```
okhan@okhan-WNJC2-MBP xrpl-py % poetry run poe test_unit

Poe => coverage run -m unittest discover tests/unit
.........................................................................................................................................................................................................................................................F.............................................................................................................................................................................................................
======================================================================
FAIL: test_to_xrpl_amm_bid (models.test_base_model.TestFromDict)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/okhan/Development/xrpl-py/tests/unit/models/test_base_model.py", line 1018, in test_to_xrpl_amm_bid
    self.assertEqual(tx.to_xrpl(), expected)
AssertionError: {'Acc[41 chars]Q', 'TransactionType': 'AMMBid', 'Flags': 0, '[674 chars]'}}]} != {'Acc[41 chars]Q', 'Asset': {'currency': 'XRP'}, 'Asset2': {'[674 chars]': 0}
  {'Account': 'r9LqNeG6qHxjeUocjvVki2XR35weJ9mZgQ',
   'Asset': {'currency': 'XRP'},
   'Asset2': {'currency': 'ETH', 'issuer': 'rpGtkFRXhgVaBzC5XCR7gyE2AZN5SN3SEW'},
-  'AuthAccounts': [{'AuthAccount': {'account': 'rNZdsTBP5tH1M6GHC6bTreHAp6ouP8iZSh'}},
?                                     ^

+  'AuthAccounts': [{'AuthAccount': {'Account': 'rNZdsTBP5tH1M6GHC6bTreHAp6ouP8iZSh'}},
?                                     ^

-                   {'AuthAccount': {'account': 'rfpFv97Dwu89FTyUwPjtpZBbuZxTqqgTmH'}},
?                                     ^

+                   {'AuthAccount': {'Account': 'rfpFv97Dwu89FTyUwPjtpZBbuZxTqqgTmH'}},
?                                     ^

-                   {'AuthAccount': {'account': 'rzzYHPGb8Pa64oqxCzmuffm122bitq3Vb'}},
?                                     ^

+                   {'AuthAccount': {'Account': 'rzzYHPGb8Pa64oqxCzmuffm122bitq3Vb'}},
?                                     ^

-                   {'AuthAccount': {'account': 'rhwxHxaHok86fe4LykBom1jSJ3RYQJs1h4'}}],
?                                     ^

+                   {'AuthAccount': {'Account': 'rhwxHxaHok86fe4LykBom1jSJ3RYQJs1h4'}}],
?                                     ^

   'BidMax': {'currency': '5475B6C930B7BDD81CDA8FBA5CED962B11218E5A',
              'issuer': 'r3628pXjRqfw5zfwGfhSusjZTvE3BoxEBw',
              'value': '35'},
   'BidMin': {'currency': '5475B6C930B7BDD81CDA8FBA5CED962B11218E5A',
              'issuer': 'r3628pXjRqfw5zfwGfhSusjZTvE3BoxEBw',
              'value': '25'},
   'Flags': 0,
   'SigningPubKey': '',
   'TransactionType': 'AMMBid'}

----------------------------------------------------------------------
Ran 455 tests in 1.256s

FAILED (failures=1)
```

Reply:
I think this might be a better solution (and it passes all the tests):
```py
def _value_to_tx_json(value: XRPL_VALUE_TYPE) -> XRPL_VALUE_TYPE:
    # IssuedCurrencyAmount and PathStep are special cases and should not be snake cased
    # and only contain primitive members
    if isinstance(value, list) and all(PathStep.is_dict_of_model(v) for v in value):
        return value
    if IssuedCurrencyAmount.is_dict_of_model(value):
        return value
    if isinstance(value, dict):
        return transaction_json_to_binary_codec_form(value)
    if isinstance(value, list):
        return [_value_to_tx_json(sub_value) for sub_value in value]
    return value
```

Reply:
Since `PathStep` is the special case and `AuthAccount` is actually behaving normally, it seemed better to keep that being the exception. `PathStep` always appears in a list (type `Path`).